dockerfiles/Dockerfile.training

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# multi-stage arguments (repeat ARG NAME below)
+ARG MOFED_VERSION=4.7-3.2.9.0
+ARG UCX_VERSION=1.8.0
+ARG OPENMPI_VERSION=4.0.4
+ARG CONDA_VERSION=4.7.10
+ARG MKL_VERSION=2020.1
+ARG NUMPY_VERSION=1.18.5
+ARG ONNX_VERSION=1.7.0
+ARG PYTORCH_VERSION=1.6.0
+
+# cuda development image for building sources
+FROM nvidia/cuda:10.2-cudnn7-devel-ubuntu18.04 as build
```

Comment:
I suggest we do not depend on Nvidia's docker images. But this is a separated topic and up to discussion.

Reply:
Since existing inference image depends on the same, I agree it's best discussed in separate PR.

Reply:
Sorry, I examined again. CUDA 10.2 and 11 are fine, they get monthly updates from Nvidia. 10.1 is not OK, but you don't need to worry :-)